### PR TITLE
Add AMQPChannel::close() method

### PR DESCRIPTION
**Pull request by pinepain**
This PR adds `AMQPChannel::close()` method. No BC changes or whatsoever.

This method was discussed many times in issues, the latest one could be found [here](https://github.com/pdezwart/php-amqp/issues/229).

### CODE REVIEW

**Comment by jippi** [comment]
This is exactly what we need too ! :)

**Comment by pinepain** [comment]
@jippi glad to hear. Feedback is always welcomed here, so if this extension misses something or has a bug you could report and issue and someone will take care of it. Sure, PRs are welcomed even more 😄.

**Comment by prolic** [comment]
Will test tomorrow.

File: amqp_channel.c [hunk 1]
```diff
@@ -1117,6 +1136,7 @@ ZEND_END_ARG_INFO()
 zend_function_entry amqp_channel_class_functions[] = {
 		PHP_ME(amqp_channel_class, __construct, 	arginfo_amqp_channel_class__construct,		ZEND_ACC_PUBLIC)
 		PHP_ME(amqp_channel_class, isConnected, 	arginfo_amqp_channel_class_isConnected,		ZEND_ACC_PUBLIC)
+		PHP_ME(amqp_channel_class, close, 			arginfo_amqp_channel_class_close,			ZEND_ACC_PUBLIC)
```
Comment (jippi) [inline amqp_channel.c]: nitpicking, indent is off for `arginfo_amqp_channel_class_close` :)

**Comment by jippi** [comment]
@pinepain thanks! :)

Currently my 2 challenges is 

- be able to break out of the blocking consume() somehow (closing the connection causes segfault :))
- being able to access the number of prefetched messages on the channel (compared to the desired number, useful when gracefully shutting down and consumer tag has been canceled)

**Comment by pinepain** [comment]
@jippi To leave blocking `consume()` you can set `read_timeout` connection option to some sane value which fits your needs and then catch timeout exception. As to segfault, can you please report issue with reproducible example please?

At this moment php-amqp uses rabbitmq-c and unfortunately it's not possible to get to prefetched messages.

File: amqp_channel.c (hunk 1 above)
Comment (pinepain) [inline amqp_channel.c]: thanks!

**Comment by jippi** [comment]
@pinepain the consume is solved exactly like that, though it feels like a hack ;) 

Will open separate issue later

**Comment by prolic** [comment]
tested and works like a charm!

**Comment by prolic** [comment]
thanks @pinepain and @lstrojny 

now release it as 1.8.0 ? :)

**Pull request merged**